### ISSUE TITLE
Fix S1128 FP: documentation tags vol. 2

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#3633]
### History

Issue with documentation tags was originally reported and fixed in #2694 with a simple reproducer:
```C#
using System; // Compliant, used by cref.

namespace SonarAnalyzer.Experiments.CSharp
{
    public enum S1128
    {
        /// <summary><see cref="DateTime"/></summary>
        DateTimeValue,
    }
}
```

### Reproducible scenario

As @mgasparel found out in [his comment](https://github.com/SonarSource/sonar-dotnet/issues/2694#issuecomment-697341005) this issue became reproducible again when:
* Using this project https://github.com/mgasparel/sonar-repro (only csproj and simple `S1128.cs` file with the reproducer)
* .NET Core 3.1 project
* SonarAnalyzer.CSharp 8.13.1 NuGet package
* `dotnet build` command

### Not reproducible scenario

This is NOT reproducible with the same setup inside Visual Studio. The FP is not reported. It also doesn't reproduce in our UTs from in-memory solution.

### Observations

It seems that `c.Node.DescendantTrivia()` does return all trivias for `CompilationUnitSyntax` under Visual Studio environment. And return empty collection under `dotnet build` context. Weird.

It also seems that it returns the trivias also under `dotnet build` context when explicitly parametrized with `descentIntoChildren` argument: `c.Node.DescendantTrivia(x => true)` but from a quick test, it seems that these comments doesn't have `HasStructure` flag set.

To make it more interesting, it seems that taking `DescendantTrivia()` from some nested node (like the `EnumDeclarationSyntax`) can return all trivias with their structure parsed and available.

There might be something environment/context dependent in Roslyn behavior (from some of it's versions). Guess would be that Roslyn doesn't parse the travias for `dotnet build` for performance reasons.

### Measurement

Way to debug this:
* Update `UnnecessaryUsings.cs` with logging to file.
* Add debug build DLLs into csproj
* `dotnet build` the reproducer project

**Comment by andrei-epure-sonarsource** [comment]
This also is reported to happen via the Azure Pipeline MSBuild task - see [community thread](https://community.sonarsource.com/t/fp-unused-using-should-be-removed/40328/2).

**Comment by andrei-epure-sonarsource** [comment]
Another report: https://community.sonarsource.com/t/s1128-raised-when-namespace-is-referenced-by-type-used-in-comment/63148

**Comment by LadislavMargai** [comment]
Do you have any update on this or a workaround?

**Comment by pavel-mikula-sonarsource** [comment]
Hi, 

we have no specific ETA to fix this, unfortunately. It might get picked up during a hardening sprint, just as with any other issue.

**Comment by markusschaber** [comment]
The same here, running sonarqube in a locally hosted gitlab instance.

**Comment by martin-strecker-sonarsource** [comment]
Probably caused by [documentation mode](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.parseoptions.documentationmode?view=roslyn-dotnet-4.6.0#microsoft-codeanalysis-parseoptions-documentationmode) set to none during the compilation. This is the same blocker as discovered by https://github.com/SonarSource/sonar-dotnet/pull/7700#issue-1825988946 and described in https://github.com/dotnet/roslyn/issues/41640 I didn't investigate this, though but it seems plausible.

**Comment by pavel-mikula-sonarsource** [comment]
@sebastien-marichal That sounds suspicious, as the root of the problem comes from the way Roslyn parses stuff => analyzer update should not change anything.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1950](https://sonarsource.atlassian.net/browse/NET-1950)

[NET-1950]: https://sonarsource.atlassian.net/browse/NET-1950?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
